### PR TITLE
feat: add maximum image file size to be processed by the thumbnailer

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
Yet another hardening around the thumbnail service.

## How Has This Been Tested?
- :hand: 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: services/thumbnails/pkg/config/config.go [hunk 1]
```diff
@@ -36,14 +36,15 @@ type FileSystemStorage struct {
...
+	FontMapFile           string            `yaml:"font_map_file" env:"THUMBNAILS_TXT_FONTMAP_FILE" desc:"The path to a font file for txt thumbnails." introductionVersion:"pre5.0"`
+	TransferSecret        string            `yaml:"transfer_secret" env:"THUMBNAILS_TRANSFER_TOKEN" desc:"The secret to sign JWT to download the actual thumbnail file." introductionVersion:"pre5.0"`
+	DataEndpoint          string            `yaml:"data_endpoint" env:"THUMBNAILS_DATA_ENDPOINT" desc:"The HTTP endpoint where the actual thumbnail file can be downloaded." introductionVersion:"pre5.0"`
+	MaxInputWidth         int               `yaml:"max_input_width" env:"THUMBNAILS_MAX_INPUT_WIDTH" desc:"The maximum width of an input image which is being processed." introductionVersion:"6.0"`
+	MaxInputHeight        int               `yaml:"max_input_height" env:"THUMBNAILS_MAX_INPUT_HEIGHT" desc:"The maximum height of an input image which is being processed." introductionVersion:"6.0"`
+	MaxInputImageFileSize uint64            `yaml:"max_input_image_file_size" env:"THUMBNAILS_MAX_INPUT_IMAGE_FILE_SIZE" desc:"The maximum file size of an input image which is being processed." introductionVersion:"6.0"`
```
Comment (mmattel) [inline services/thumbnails/pkg/config/config.go:49]: We need to add the unit of the value in the description. int and uint64 is just the computational range.

**Example:**
`The maximum file size of an input image which is being processed.` -->
`The maximum file size in bytes of an input image that will be processed.`

**Review by mmattel — APPROVED** [review]
LGTM from a docs pov 👍

**Review by rhafer — CHANGES_REQUESTED** [review]
I think this need a Changelog. Looks good otherwise.

**Comment by DeepDiver1975** [comment]
> I think this need a Changelog. Looks good otherwise.

done - added to other changelog item which falls into the same topic

File: services/thumbnails/pkg/config/config.go [hunk 2]
```diff
@@ -36,14 +36,15 @@ type FileSystemStorage struct {
...
+	TransferSecret        string            `yaml:"transfer_secret" env:"THUMBNAILS_TRANSFER_TOKEN" desc:"The secret to sign JWT to download the actual thumbnail file." introductionVersion:"pre5.0"`
+	DataEndpoint          string            `yaml:"data_endpoint" env:"THUMBNAILS_DATA_ENDPOINT" desc:"The HTTP endpoint where the actual thumbnail file can be downloaded." introductionVersion:"pre5.0"`
+	MaxInputWidth         int               `yaml:"max_input_width" env:"THUMBNAILS_MAX_INPUT_WIDTH" desc:"The maximum width of an input image which is being processed." introductionVersion:"6.0"`
+	MaxInputHeight        int               `yaml:"max_input_height" env:"THUMBNAILS_MAX_INPUT_HEIGHT" desc:"The maximum height of an input image which is being processed." introductionVersion:"6.0"`
+	MaxInputImageFileSize string            `yaml:"max_input_image_file_size" env:"THUMBNAILS_MAX_INPUT_IMAGE_FILE_SIZE" desc:"The maximum file size of an input image which is being processed. Usable common abbreviations: [KB, KiB, MB, MiB, GB, GiB, TB, TiB, PB, PiB, EB, EiB], example: 2GB." introductionVersion:"6.0"`
```
[the remark this answers is not in the document]
Comment (DeepDiver1975) [reply services/thumbnails/pkg/config/config.go:49]: > Do we have some experience how the file size maps to memory consumption when generating a thumbnail?

not really - and this is rather complicated...

From my understanding the image dimensions are more critical to this. A rather small file can hold an image with gigantic dimensions which can blow up the thumbnailer pretty easy ....



> Just as a "unrelated" question: are there any plans to limit the concurrency for generating thumbnails, too?

I did not look into this yet. Feel free to add this to https://github.com/owncloud/enterprise/issues/6388
THX

**Review by rhafer — APPROVED** [review]

**Pull request merged**